### ISSUE TITLE
group_recorder interval=-1 produces no output

### ISSUE THREAD

**Issue by trevorhardy** [gridlab-d/gridlab-d#1085]
- [x] I have not found an open issue describing the same problem.
- [x] I went through "CONTRIBUTING.MD" before deciding to submit the issue.
- [x] While building GridLAB-D, I followed all steps as detailed in GridLAB-D Wiki.

------------------------------------------------------------------------------------------------------------

## Please describe what was the cause of the issue?
group_recorder objects support interval parameter values of -1 (record value on change) but provide write no value to the output file beyond the header lines AND produce no error on the console stating the group_recorder is doing nothing.

## If possible, please provide any screenshots and other supporting information.
Attached is a two-house model demonstrating the bug.
[Two house.zip](https://github.com/gridlab-d/gridlab-d/files/1434949/Two.house.zip)

**Comment by dchassin** [comment]
I have noticed that sometimes interval=-1 does not cause the clock to advance by itself.  You must have something else to drive the clock forward and then the recorder will pick up changes that it can output.

**Comment by ftuffner** [comment]
This strikes me as a fundamental question -- _should_ group_recorder drive the simulation forward with a -1 interval?  My assumption was always -1 for recorder/group_recorder was reactive (something else has to drive it).  Only explicit positive integers would drive the simulation forward.

**Comment by dchassin** [comment]
I agree with Frank that this behavior is correct given the meaning of "interval -1" in this context.  But it certainly is unexpected and confusing to the novice user. In fact it trips me up sometimes.  

When I teach about the tape module, I'm always careful to point out that there is a difference between passive recording and active recording.  Passive recording (like -1) does not change the overall behavior of the simulation, while active recording (like positive intervals) can affect the simulation's overall behavior.  A good example is what happens when you turn on interpolation in the climate module.  With interval -1, the weather updates hourly with no interpolation unless something else causes subhourly sampling.  With a positive interval < 1h, the weather will be updated subhourly, which gives a potentially different result.  In my humble opinion, it's not clearly right or wrong according to our definitions of how the objects involved behave.  It's just different.
